### PR TITLE
initial pddl parser use and implementation!

### PR DESCRIPTION
**Pull request by NishanthJKumar**

### CODE REVIEW

File: lisdf/parsing/parse_pddl.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+from typing import Dict, List, Set, Tuple
+
+import pddlpy
+
+
+def parse_pddl(
+    domain_file_path: str, problem_file_path: str
+) -> Tuple[List[List[str]], List[List[str]], Dict[str, Set[str]]]:
```
Comment (williamshen-nz) [inline lisdf/parsing/parse_pddl.py:8]: I personally would prefer if this were wrapped in a dataclass/namedtuple or something. That's probably a bit too much effort for now though.

Could we make the typing a bit more readable? e.g.

```
Predicate = str
Type = str
Object = str

# First element in an atom is the predicate name, the rest are argument objects
Atom = List[Union[Predicate, Object]]
State = List[Atom]

-> Tuple[State, State, Dict[Type, Set[Object]]
```

File: lisdf/parsing/parse_pddl.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    goal_atoms_strs = [atom.predicate for atom in goal_atoms]
+    obj_to_type_dict = dom_prob.worldobjects()
+    type_to_object_dict: Dict[str, Set[str]] = {}
+    for obj in obj_to_type_dict.keys():
+        if type_to_object_dict.get(obj_to_type_dict[obj]) is None:
+            type_to_object_dict[obj_to_type_dict[obj]] = set([obj])
+        else:
+            type_to_object_dict[obj_to_type_dict[obj]].add(obj)
+
+    return (init_atoms_strs, goal_atoms_strs, type_to_object_dict)
```
Comment (williamshen-nz) [inline lisdf/parsing/parse_pddl.py:37]: nit: `( )` unnecessary here

File: tests/test_parsing/test_parse_pddl.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+
+def test_parse_pddl():
+    init_atoms, goal_atoms, type_to_obj_dict = parse_pddl(
+        "examples/pddl/clutter.pddl", "examples/pddl/clutter_prob01.pddl"
+    )
+    for atom in init_atoms:
+        # Test that the 0th element of the atoms, which should represent
+        # the predicate names, is in the list of known predicate names
+        assert atom[0] in ["on-surface", "arm-empty", "holding", "on"]
```
Comment (williamshen-nz) [inline tests/test_parsing/test_parse_pddl.py:11]: Can you add check that none of the other elements in the list are in the predicate names. Something like

```
assert not any(elem in predicate_names for elem in atom[1:])
```

**Review by williamshen-nz — COMMENTED** [review]
Looks great! Just a few minor comments

**Review by williamshen-nz — APPROVED** [review]
:shipit: Thanks for making the changes!

**Comment by NishanthJKumar** [comment]
Closing in favor of #12 , which seems to include its own, better PDDL parsing utility!

**Pull request closed without merging**